### ISSUE TITLE
[BUG] vsphere 8 using rke2/k3s - if vsphere DRS maxVcpusPerCore limit is reached on host, rancher does not clean up vsphere VMs

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#44212]
**Rancher Server Setup**
- Rancher version: v2.8.0
- Installation option (Docker install/Helm Chart): docker install
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): v1.27.8
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: any (tested on 1.27.8+rke2r1)
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): rancher provisioned via vsphere node driver

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
tested as administrator on both rancher and vsphere

**Describe the bug**

if vsphere DRS `maxVcpusPerCore` is reached, rancher doesn't clean up the VMs that fail to start in vsphere. 

**To Reproduce**

in vsphere:
* in your datacenter cluster, set `maxVcpusPerCore` to a low number relative to the physical CPUs available i.e. 1 or 2 
in rancher:
* provision a cluster using vsphere node driver + credentials that will exceed the limit set on DRS
* wait for the cluster to start provisioning

**Result**
After the limit is hit, the nodes fail to fully provision. This would be expected, however the VMs are left around on the vsphere setup. 

**Expected Result**

Typically node failures are deleted from the vsphere setup before rancher retries creating a VM

**Screenshots**

**Additional context**

this may happen on vsphere 7 as well, but not tested. 
not tested on rke1

**Comment by jiaqiluo** [comment]
waiting for cutting a new rancher-machine RC and bring it into rancher/rancher

**Comment by jiaqiluo** [comment]
PR is merged, now waiting for new r/r image

**Comment by jiaqiluo** [comment]
the fix can be validated on the latest v2.9-head

**Comment by slickwarren** [comment]
Is there a way to get the error from VMware / vsphere for any generic error? I tried using a similar testing approach, but not the same as the original ticket, and got a new error in vsphere that doesn't show up in rancher on v2.9-head (92f15949d71efb11baf63c878cc2f64bcd25b1e8): 
```
 Invalid virtual machine configuration. A virtual IOMMU should always be enabled for a VCPU count of more than 128. 
```

```
 The total number of virtual CPUs present or requested in virtual machines' configuration has exceeded the limit on the host: 192. 
```

I was able to do this by:
giving a rancher downstream cluster an extremely large cpu count
using a reasonable cpu count for nodes, but having more nodes * cpu than the vsphere setup is configured for.
